docs/docsite/rst/dev_guide/developing_program_flow_modules.rst

```diff
@@ -466,18 +461,65 @@ this. If a module has to use this on its own, it should instantiate an
 
 .. versionadded:: 2.1
 
+
 _ansible_version
 ^^^^^^^^^^^^^^^^
 
-This parameter passes the version of Ansible that runs the module. To access
-it, a module should instantiate an `AnsibleModule` and then retrieve it
-from :attr:`AnsibleModule.ansible_version`. This replaces
-:attr:`ansible.module_utils.basic.ANSIBLE_VERSION` from
-:ref:`module_replacer`.
+This argument passes the version of Ansible to the module. To access it, a module should instantiate the ``AnsibleModule`` method and then retrieve the version from :attr:`AnsibleModule.ansible_version`. This replaces :attr:`ansible.module_utils.basic.ANSIBLE_VERSION` from :ref:`module_replacer`.
 
 .. versionadded:: 2.1
 
 
+_ansible_module_name
+^^^^^^^^^^^^^^^^^^^^
+
+Format: <str>
+
+This argument passes the information to modules about their name.
+
+
+_ansible_string_conversion_action
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+This argument provides instructions about what modules should do after the values of the user-specified module parameters are converted to strings.
+
+
+_ansible_keep_remote_files
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Format: <bool>
+
+This argument provides instructions that modules must be ready if they need to keep the remote files.
+
+
+_ansible_socket
+^^^^^^^^^^^^^^^
+This argument provides modules with a socket for persistent connections.
+
+
+_ansible_shell_executable
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Format: <bool>
+
+This argument ensures that modules use the designated shell executable.
+
+
+_ansible_tmpdir
+^^^^^^^^^^^^^^^
+
+Format: <str>
+
+This argument provides instructions to modules that all commands must use the designated temporary directory, if created.
```

Comment:
this is somewhat of a mix, part of it is configurable (root path) but the end directory is generated randomly for each task/host so while you can point at a config, ti won't control the full value.